devel/grpc/Portfile

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem 1.0
+PortGroup  github 1.0
+PortGroup  cxx11 1.1
+
+github.setup    grpc grpc 1.9.1 v
+categories      devel
+maintainers     nomaintainer
+license         Apache-2
+description     A high performance, open-source universal RPC framework.
+homepage        https://grpc.io
+platforms       darwin
+
+long_description \
+    Remote Procedure Calls (RPCs) provide a useful abstraction \
+    for building distributed applications and services. The \
+    libraries in this repository provide a concrete \
+    implementation of the gRPC protocol, layered over HTTP/2. \
+    These libraries enable communication between clients and \
+    servers using any combination of the supported languages.
+
+checksums       rmd160  90709202459e8bfd71762ac3cafd85fe503b10d4 \
+                sha256  43c1e96886d6ee3fbb2978f645400a92df989d37c80f2408f6730cfc1e05875b
+
+patchfiles patch-Makefile.diff
+
+depends_lib-append \
+    port:zlib \
+    port:protobuf3-cpp \
+    port:c-ares \
+    port:openssl
+
+depends_build-append \
+    port:pkgconfig \
+    port:autoconf \
+    port:automake \
+    port:libtool
```

Comment:
I'm super confused. You are declaring dependency on autoconf, automake, etc. and then not using any of those anywhere, but manually patching some `Makefile` and not even running the configure script? Also, there's cmake support in the software, so I'm even more confused about what is going one.

Reply:
I see the potential for confusion. I declared the build dependencies that the [gRPC Install](https://github.com/grpc/grpc/blob/master/INSTALL.md) suggests.  It's not clear if they're actually being used by `make`.

The suggested build systems for macOS is `make` on the project supplied `Makefile`.  As such there is no configure script. I tried the CMake build system but it needed patching as well in order to find dependencies and iirc there was an issue with its compiled dylibs too.

Reply:
Is there a good way to determine if autoconf, automake, and libtool are really necessary? I tried trace mode and I didn't see any complaint that they weren't necessary.